ui/src/app/services/profile-context.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {useEffect} from 'react';
+
+import {Profile} from 'generated/fetch';
+
+import {profileApi} from 'app/services/swagger-fetch-clients';
+import {apiCallWithConflictRetries} from 'app/utils/retry';
+import {useStore} from 'app/utils/stores';
+import {atom} from 'app/utils/subscribable';
+
+interface ProfileStore {
+  profile?: Profile;
+}
+
+const reactProfileStore = atom<ProfileStore>({});
+
+const ProfileContext = React.createContext({
+  profile: {},
+  reload: () => {}
+});
+
+export const ProfileProvider = ({children}) => {
+  useStore(reactProfileStore);
+
+  const getProfile = async() => {
+    const profile = await apiCallWithConflictRetries(() => profileApi().getMe());
+    reactProfileStore.set({...reactProfileStore.get(), profile: profile});
+    return profile;
+  };
+
+  useEffect(() => {
+    async function getProfileWrapper() {
+      await getProfile();
+    }
+
+    getProfileWrapper();
```

Comment:
Ah, yeah, for _this_ store ideally you'd fetch only on first render and upon explicit call to `reload`. It will vary per store - for instance, `cdrVersionStore` you'd want to fetch once per session. I can clarify in comments for sure